### ISSUE TITLE
Using flathub provided OpenJDK SDK

### ISSUE THREAD

**Issue by CC-Hsu** [flathub/org.prismlauncher.PrismLauncher#86]
Hi,

There are many Flathub packages that are rely on [OpenJDK SDK](https://github.com/flathub/org.freedesktop.Sdk.Extension.openjdk), such as [Shattered Pixel Dungeon](https://github.com/flathub/com.shatteredpixel.shatteredpixeldungeon/blob/master/com.shatteredpixel.shatteredpixeldungeon.yaml#L7) or [Unciv](https://github.com/flathub/io.github.yairm210.unciv/blob/master/io.github.yairm210.unciv.yaml#L5).

And there are many supported OpenJDK versions provided in Flathub

It would be great if PrismLauncher can also include this support, too.

Best Regards.

**Comment by Scrumplex** [comment]
We used to ship these but decided to drop them in favor of Prism Launcher's built-in Java downloader.

**Comment by CC-Hsu** [comment]
Hi, Scrumplex,

Thanks for your reply, I can understand that these java-based games bind a specific OpenJDK version in this way,
while PrismLauncher aims to provide flexibility on choosing different version Java for different version MC.

I remind that Flatpak VS Code can supports multiple SDK Flatpak packages without any pre-specified SDK package name and version; but I can only find the below line and not sure whether this is the exact line for this magic.
https://github.com/flathub/com.visualstudio.code/blob/master/com.visualstudio.code.yaml#L4

I guess maybe this might be a way, too.

**Comment by Scrumplex** [comment]
Previously we had to copy the contents from the OpenJDK SDKs to our Flatpak, which made our application significantly larger to download. Note that we currently would need to ship three JREs

**Comment by CC-Hsu** [comment]
Yeah, I can understand shipping built-in Java runtime is a nightmare.

But I find that VS Code Flatpak is shipped without pre-packaged language SDKs:
it can be installed alone without JDK, golang, rust, .Net ...etc included,
and it recognize a runtime/compiler under `/lib/sdk/` directory as someone install any `org.freedesktop.sdk.*` package.

Would that be a possible way to make Flatpak OpenJDK SDKs as possible runtimes for the PrismLauncher flatpak?
